code/modules/mob/living/basic/icemoon/ice_whelp/ice_whelp_abilities.dm

```diff
@@ -0,0 +1,33 @@
+/datum/action/cooldown/ice_breathe
+	name = "Ice breathe"
+	desc = "Fire a cold line of fire towards the enemy!"
+	button_icon = 'icons/effects/magic.dmi'
+	button_icon_state = "fireball"
+	cooldown_time = 4 SECONDS
+	melee_cooldown_time = 0
+	click_to_activate = TRUE
+	///the range of fire
+	var/fire_range = 4
+
+/datum/action/cooldown/ice_breathe/Activate(atom/target_atom)
+	var/turf/target_fire_turf = get_ranged_target_turf_direct(owner, target_atom, fire_range)
+	var/list/burn_turfs = get_line(owner, target_fire_turf) - get_turf(owner)
+	dragon_fire_line(owner, burn_turfs, frozen = TRUE)
+	StartCooldown()
+	return TRUE
+
+/datum/action/cooldown/ice_breathe/all_directions
```

Comment:
Ditto, also there doesn't seem to be a good reason to subtype this from `ice_breath`, you don't use any of it.